handbook/singularity.md

```diff
@@ -0,0 +1,40 @@
+# Singularity
+
+On occasion, the operating system and Rstudio installations that are available via [FASRC VDI Apps](https://docs.rc.fas.harvard.edu/kb/vdi-apps/) are not compatible with some R packages. A viable way to surmount package incompatibility issues is using docker containers. The [Rocker Project](https://rocker-project.org/images/) images run R and Rstudio containerized installations that come in various flavors. Three of the most popular rocker image builds are: `rocker/tidyverse`, `rocker/verse`, and `rocker/geospatial`. The official overview of these images is found [here](https://rocker-project.org/images/versioned/rstudio.html#overview). 
+
+## Downloading a rocker/docker image with singularity
+
+To work on a containerized Rstudio in FASSE, first identify the rocker distribution that best fits your needs. For the purpose of this tutorial, we will work with `rocker/geospatial:3.6.0`.
+
+Once logged in to FASSE, create an `images` folder in your user directory and run
+
+```
+cd images
+singularity pull docker://rocker/geospatial:3.6.0
+```
+
+Once the download has completed you should be able to see the file `geospatial_3.6.0.sif`.
+
+## Running a rocker/docker image with singularity
+
+Go to [FASRC VDI Apps](https://docs.rc.fas.harvard.edu/kb/vdi-apps/) and open a Remote Desktop session. Within the Remote Desktop, open a new terminal and inside the images folder run  
+
+```
+cd images
+singularity shell geospatial_3.6.1.sif
+```
+
+Inside the running container, copy the following lines. You can chose the password of your preference.
+
+```
+export PASSWORD=geospatial
```

Comment:
Another useful line to add is RSTUDIO_SESSION_TIMEOUT='0'
which makes sure you don't lose your session (it sometimes times out after 30min or an hour or something)

Reply:
I am not convinced either of the password requirement. I know a workaround to avoid including password in docker but have not tried one in singularity.

Reply:
I will investigate and include a non-password command

Reply:
I didn't know about the timeout! it has not happened to me yet!

Reply:
I have been using without a password, but as far as what is best practice, I don't know. If there would be any way for someone else to access the rserver container on the same node this is bad--otherwise I think it doesn't matter.
My command to run without password is to start singularity shell file.sif, then type rserver, load firefox localhost, and it works.